src/Shared/PackageManagers/CPANProjectManager.cs

```diff
@@ -22,6 +22,24 @@ public CPANProjectManager(string destinationDirectory) : base(destinationDirecto
         {
         }
 
+        /// <summary>
+        /// Check if the package exists in the respository.
+        /// </summary>
+        /// <param name="purl">The PackageURL to check.</param>
+        /// <param name="useCache">If cache should be used.</param>
+        /// <returns>True if the package is confirmed to exist in the repository. False otherwise.</returns>
+        public override async Task<bool> PackageExists(PackageURL purl, bool useCache = true)
+        {
+            Logger.Trace("PackageExists {0}", purl?.ToString());
+            if (purl is null || purl.Name is null || purl.Type is null)
```

Comment:
nit: do we need to check if `purl.Name` and/or `purl.Type` are also empty strings? Also do we need to check if `purl.Type` is `null` if we're not using it in this function?

(Looks like this comment can be applied to most project manager classes, so feel free to do what you'd like with this comment 😀)

Reply:
I think I was overzealous in checking Type and that can in fact be removed.

If it is an empty string the later call will likely fail. So it probably does make sense to check that here as well as the purpose of this check is to shortcircuit actually making a HTTP call for something we know won't return any results.